### PR TITLE
Feature: focus tab on link to tab

### PR DESCRIPTION
**Pull request by SallyMcGrath**
## What does this change?

Module: all
Week(s): all

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.MD)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works
- [x] My changes follow our [Style Guide](https://curriculum.codeyourfuture.io/guides/code-style-guide)

## Description

-Inspects the URL fragment when the component mounts in the connectedCallback() method.
-Created a new method, _handleFragmentNavigation(), to simulate a click event on the tab specified in the URL fragment.
-Listening to hashchange events so that the component will respond to changes in the URL fragment even after the component has mounted.

No style changes yet in this pr just function and a content sample on fundamentals/prep

### CODE REVIEW

**Comment by SallyMcGrath** [comment]
ugh sorry, I will ff this as well, and find out where the squash merge button is

**Comment by SallyMcGrath** [comment]
the broken youtube is unrelated to tabs -- the issue is it's rendering as an html template (with p tags and all) and so the split function can't find the id properly.

**Comment by SallyMcGrath** [comment]
Oh I also wrote a new style and redid the aria

File: content/fundamentals/prep/index.md [hunk 1]
```diff
@@ -44,12 +44,12 @@ GitHub is for developers. It's how you will work on code with other developers a
 Your GitHub account name should look something like this: https://github.com/SallyMcGrath
 {{</note>}}
 
-When prospective employers are looking at your GitHub portfolio, **you need them to know who you are**: not your online identity, but the name you put on your job application. Don’t use cute handles on your GitHub, even though some mentors do. They are not applying for entry level developer roles.
+When prospective employers are looking at your Github portfolio, **you need them to know who you are**: not your online identity, but the name you put on your job application. Don’t use cute handles on your Github, even though some mentors do. They are not applying for entry level developer roles.
```
Comment (illicitonion) [inline content/fundamentals/prep/index.md:47]:
```suggestion
When prospective employers are looking at your GitHub portfolio, **you need them to know who you are**: not your online identity, but the name you put on your job application. Don’t use cute handles on your GitHub, even though some mentors do. They are not applying for entry level developer roles.
```

How did these get undone? :D

File: content/fundamentals/prep/index.md [hunk 2]
```diff
@@ -44,12 +44,12 @@ GitHub is for developers. It's how you will work on code with other developers a
...
 
-To find out more about what GitHub is used for, watch the video in the "What is GitHub" tab above.
+Watch [What is Github](#prep-1)
```
Comment (illicitonion) [inline content/fundamentals/prep/index.md:49]:
```suggestion
Watch [What is GitHub](#prep-1)
```

File: assets/styles/04-components/tabs.scss [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
-  margin-bottom: var(--theme-spacing--6);
+  --outset: calc(-1 * var(--theme-spacing--gutter));
+  --tabline: 1px;
+  margin: 0 var(--outset) var(--theme-spacing--6);
   &__list {
     display: flex;
   }
-  &__link {
+  &__tab {
+    font-family: var(--theme-font--display);
```
Comment (illicitonion) [inline assets/styles/04-components/tabs.scss:10]:
```suggestion
    cursor: hand;
    font-family: var(--theme-font--display);
```

Comment (illicitonion) [reply assets/styles/04-components/tabs.scss:10]: Not sure if we want some handy "fake-link" style we can inherit (I don't know scss, sorry!), but we should apply this to the clickable thing

**Review by illicitonion — APPROVED** [review]

**Pull request merged**